tcmba/spiders/obras.py

```diff
@@ -0,0 +1,114 @@
+from datetime import date, datetime
+
+from scrapy import Request, Spider
+
+from tcmba.items import ConstructionItem
+from tcmba.spiders.helpers import format_year
+
+
+class ConstructionsSpider(Spider):
+    name = "obras"
+    url = "https://www.tcm.ba.gov.br/portal-da-cidadania/obras/?municipio=2910800"
+    current_year = date.today().year
+
+    def __init__(self, *args, **kwargs):
+        super(ConstructionsSpider, self).__init__(*args, **kwargs)
+        if hasattr(self, "ano_inicial"):
+            try:
+                int(self.ano_inicial)
+                self.ano_inicial = int(format_year(self.ano_inicial))
+            except ValueError:
+                raise Exception("Você precisa informar um ano válido.")
+        else:
+            self.ano_inicial = date.today().year
+
+    def start_requests(self):
+        year = self.ano_inicial
+        while year <= self.current_year:
+            # 129 é o código de entidade da Prefeitura Mun. de Feira de Santana
+            url = f"{self.url}&ano={year}&entidade=129&pesquisar=Pesquisar"
+            year += 1
+            yield Request(url, self.parse)
+
+    def parse(self, response):
+        rows = response.css("table#tabelaResultado tbody tr")
+
+        for row in rows:
+            detail_url = (
+                row.css(".btn-acao::attr(onclick)")
+                .re_first(r"location.=\s*(.*)")
+                .replace("'", "")
+            )
+
+            company_content = row.css("a.btn-popover::attr(data-content)")
```

Comment:
```suggestion
            company_content = row.css("a.btn-popover::attr(data-content) td")
```

```suggestion
                    "name": company_content.re_first(r"\D+"),
```

E aí fazer o match. Seria mais seguro pra dar match com as colunas.

Reply:
Nesse ponto, não sei se vai funcionar.. Pois este conteúdo está como um data-content dentro da tag <a>. É usado no popover do Bootstrap

![image](https://user-images.githubusercontent.com/14143674/135718758-45563d2f-13ba-4522-8ced-2f2d9f046844.png)

Reply:
resolvido com o help da Ana

![image](https://user-images.githubusercontent.com/14143674/135722096-49c94a1a-a3c5-44e5-a8b2-1035783f6113.png)